go/tools/generate_test_main.go

```diff
@@ -39,6 +59,29 @@ type Cases struct {
 	HasTestMain      bool
 	Version17        bool
 	Version18OrNewer bool
+	Cover            []coverInfo
```

Comment:
It is actually copied from $GOROOT/src/cmd/go/test.go. I suggest we keep it so that it will be easier to compare and update with the reference impl. Also, the map is needed because we need to track cover vars for each file.
Thanks!